core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1102,6 +1109,9 @@ object KafkaConfig {
       .define(ShareGroupMaxSizeProp, SHORT, Defaults.SHARE_GROUP_MAX_SIZE, between(10, 1000), MEDIUM, ShareGroupMaxSizeDoc)
       .define(ShareGroupAssignorsProp, LIST, Defaults.SHARE_GROUP_ASSIGNORS, null, MEDIUM, ShareGroupAssignorsDoc)
       .defineInternal(ShareGroupPersisterClassNameProp, STRING, Defaults.SHARE_GROUP_PERSISTER_CLASS_NAME, new ConfigDef.NonNullValidator(), MEDIUM, ShareGroupPersisterClassNameDoc)
+      .defineInternal(ShareGroupStateTopicReplicationFactorProp, SHORT, Defaults.SHARE_GROUP_STATE_TOPIC_REPLICATION_FACTOR, atLeast(1), HIGH, ShareGroupStateTopicReplicationFactorDoc)
```

Comment:
These should not be internal configs. There are 4 configs in the KIP: `share.coordinator.state.topic.num.partitions`, `share.coordinator.state.topic.replication.factor`, `share.coordinator.state.topic.segment.bytes` and `share.coordinator.state.topic.min.isr`. While I understand why the names in this PR don't agree with the KIP (they changed towards the end of Jun's comments), it is probably worth adding the minISR config in this PR. I think it's probably worth taking the name change now too.